key.go

```diff
@@ -232,6 +233,23 @@ func (k Key) IsTopLevel() bool {
 	return len(k.List()) == 1
 }
 
+// MarshalJSON implements the json.Marshaler interface,
+// keys are represented as JSON strings
+func (k Key) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + k.String() + `"`), nil
```

Comment:
You might want to do some string escaping here, otherwise some keys might produce broken json

Reply:
very good point, will add

Reply:
has this been addressed?

Reply:
Any reason to not just call `json.Marshal(k.String())`?